app/gui/src/project-view/components/visualizations/TableVisualisationTooltip.ts

```diff
@@ -41,20 +40,25 @@ export class TableVisualisationTooltip implements ITooltipComp {
       return `<div style="display: inline-block; width: 10px; height: 10px; border-radius: 50%; background-color: ${color}; margin-left: 5px;"></div>`
     }
 
-    const dataQualityTemplate = `
-            <div style="display: ${getDisplay(params.numberOfNothing)};">
-                Nulls/Nothing: ${getPercentage(params.numberOfNothing)}% ${createIndicator(+getPercentage(params.numberOfWhitespace))}
-            </div>
-            <div style="display: ${getDisplay(params.numberOfWhitespace)};">
-                Trailing/Leading Whitespace: ${getPercentage(params.numberOfWhitespace)}% ${createIndicator(+getPercentage(params.numberOfWhitespace))}
-            </div>
-        `
+    const getDataQualityTemplate = () => {
+      let template = ''
+      params.dataQualityMetrics.forEach((obj) => {
+        const key = Object.keys(obj)[0]
+        const value = key ? obj[key] : null
+        if (key && value) {
```

Comment:
`value` truthiness check here seems to make `getDisplay` redundant.

Reply:
Can we get an `else` branch that does some `console.log` telling us that a metric format is malformed?

This was one of the problems with debugging visualizations - there are similar situations where it just works or just doesn't work but doesn't tell us why. Let's try to log these bad conditions so that we can actually figure out what is wrong. IMHO every `if` that checks a required condition should have an else branch logging that it is violated.